tools/update-metadata.lua

```diff
@@ -65,38 +48,314 @@ local function getLocales()
+    <color type="primary" scheme_preference="dark">#303030</color>
+  </branding>
+
+  <requires>
+    <display_length>360</display_length>
+  </requires>
+
+  <supports>
+    <control>pointing</control>
+    <control>keyboard</control>
+    <control>gamepad</control>
+  </supports>
+
+  <recommends>
+    <memory>1024</memory>
+    <display_length compare="ge">600</display_length>
+    <internet>always</internet>
+    <control>touch</control>
+  </recommends>
+
+  <provides>
+    <mediatype>application/epub+zip</mediatype>
+    <mediatype>application/fb2</mediatype>
+    <mediatype>application/fb3</mediatype>
+    <mediatype>application/msword</mediatype>
+    <mediatype>application/oxps</mediatype>
+    <mediatype>application/pdf</mediatype>
+    <mediatype>application/rtf</mediatype>
+    <mediatype>application/tcr</mediatype>
+    <mediatype>application/vnd.amazon.mobi8-ebook</mediatype>
+    <mediatype>application/vnd.comicbook+tar</mediatype>
+    <mediatype>application/vnd.comicbook+zip</mediatype>
+    <mediatype>application/vnd.ms-htmlhelp</mediatype>
+    <mediatype>application/vnd.openxmlformats-officedocument.wordprocessingml.document</mediatype>
+    <mediatype>application/vnd.palm</mediatype>
+    <mediatype>application/x-cbz</mediatype>
+    <mediatype>application/x-chm</mediatype>
+    <mediatype>application/x-fb2</mediatype>
+    <mediatype>application/x-fb3</mediatype>
+    <mediatype>application/x-mobipocket-ebook</mediatype>
+    <mediatype>application/x-tar</mediatype>
+    <mediatype>application/xhtml+xml</mediatype>
+    <mediatype>application/xml</mediatype>
+    <mediatype>application/zip</mediatype>
+    <mediatype>image/djvu</mediatype>
+    <mediatype>image/gif</mediatype>
+    <mediatype>image/jp2</mediatype>
+    <mediatype>image/jpeg</mediatype>
+    <mediatype>image/jxr</mediatype>
+    <mediatype>image/png</mediatype>
+    <mediatype>image/svg+xml</mediatype>
+    <mediatype>image/tiff</mediatype>
+    <mediatype>image/vnd.djvu</mediatype>
+    <mediatype>image/vnd.ms-photo</mediatype>
+    <mediatype>image/x-djvu</mediatype>
+    <mediatype>image/x-portable-arbitrarymap</mediatype>
+    <mediatype>image/x-portable-bitmap</mediatype>
+    <mediatype>text/html</mediatype>
+    <mediatype>text/plain</mediatype>
+  </provides>
+
+  <launchable type="desktop-id">rocks.koreader.KOReader.desktop</launchable>
+
+  <content_rating type="oars-1.1"/>
+]],
+}
+
+local function summary(lang)
+    local lang = lang or "en"
+    _.changeLang(lang)
+    return _(metadata.summary)
+end
+
+local function htmlDescription(lang)
+    local lang = lang or "en"
+    local desc = metadata.desc
+    local t = {}
+    _.changeLang(lang)
+    for i, v in ipairs (desc.paragraphs) do
+        table.insert(t, "<p>" .. util.htmlEscape(_(v)) .. "</p>")
+    end
+
+    table.insert(t, "<ul>")
+    for i, v in ipairs(desc.highlights) do
+        table.insert(t, "  <li>" .. _(v) .. "</li>")
```

Comment:
Anything within tags, right? For both xml and html metadata?.

Reply:
Matters less for HTML but yes. :+1:

Reply:
actually I'm not sure if this is what you have in mind.

```diff
diff --git a/platform/common/koreader.metainfo.xml b/platform/common/koreader.metainfo.xml
index 31f4a3892..f84868466 100644
--- a/platform/common/koreader.metainfo.xml
+++ b/platform/common/koreader.metainfo.xml
@@ -96,7 +96,7 @@
     <p>KOReader is an ebook reader optimized for e-ink screens. It can open many formats and provides advanced text adjustments.</p>
     <p>See below for a selection of its many features:</p>
     <ul>
-      <li>Supports both fixed page formats (PDF, DjVu, CBT, CBZ) and reflowable e-book formats (EPUB, FB2, Mobi, DOC, CHM, TXT, HTML). Scanned PDF/DjVu documents can be reflowed. Special flow directions for reading double column PDFs and manga.</li>
+      <li>Supports both fixed page formats (PDF, DjVu, CBT, CBZ) and reflowable e-book formats (EPUB, FB2, Mobi, DOC, CHM, TXT, HTML). Scanned PDF&#47;DjVu documents can be reflowed. Special flow directions for reading double column PDFs and manga.</li>
       <li>Multi-lingual user interface optimized for e-ink screens. Highly customizable reader view with complete typesetting options. Multi-lingual hyphenation dictionaries are bundled in.</li>
       <li>Non-Latin script support for books, including the Hebrew, Arabic, Persian, Russian, Chinese, Japanese and Korean languages.</li>
       <li>Unique Book Map and Page Browser features to navigate your book.</li>
```

But that's maybe because I don't understand why we need to escape strings in here. We're serializing everything to a giant string and dump that in binary mode.

Reply:
It's because someone might translate it as `<PDF, DjVu, CBT, CBZ>`, or as `PDFs & manga`. Or we might innocently do something like that ourselves in the future. That we use a giant string is the problem description.

Reply:
It's not watertight, but if you escape `&`, `<` and `>` chances of anything ever breaking are negligible, while if you don't it's a near certainty.

Reply:
But then we can escape the whole document together just before writing it to a file?

If not how we're going to handle unscaped string literals. For instance:

```
  <url type="homepage">https://koreader.rocks/</url>
  <url type="bugtracker">https://github.com/koreader/koreader/issues</url>
  <url type="faq">https://koreader.rocks/user_guide</url>
  <url type="translate">https://hosted.weblate.org/engage/koreader/</url>
  <url type="vcs-browser">https://github.com/koreader/koreader</url>
  <url type="contribute">https://koreader.rocks/doc/</url>
```

While, an escaped url looks like `<image>https:&#47;&#47;github.com&#47;koreader&#47;koreader-artwork&#47;raw&#47;master&#47;koreader-menu-framed.png</image>`

Reply:
> But then we can escape the whole document together just before writing it to a file?

No, that is impossible. Simply escape that which needs to be escaped (i.e., the translated text).